presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -175,7 +175,7 @@ public static long castToInteger(@SqlType(StandardTypes.DOUBLE) double value)
             return Ints.checkedCast((long) MathFunctions.round(value));
         }
         catch (IllegalArgumentException e) {
-            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, e);
+            throw new PrestoException(NUMERIC_VALUE_OUT_OF_RANGE, "Out of range for integer: " + value, e);
```

Comment:
Since we show the value here, saying "from double" is redundant. Tracking errors back to the original query position is a much bigger project (there is an issue for that).